src/components/transport_manager/src/transport_adapter/transport_adapter_impl.cc

```diff
@@ -756,25 +748,53 @@ void TransportAdapterImpl::RemoveFinalizedConnection(
     const DeviceUID& device_handle, const ApplicationHandle& app_handle) {
   const DeviceUID device_uid = device_handle;
   LOG4CXX_AUTO_TRACE(logger_);
-  sync_primitives::AutoWriteLock lock(connections_lock_);
+  connections_lock_.AcquireForWriting();
```

Comment:
@fronneburg why did you remove AutoLock in that case?

Reply:
@LuxoftAKutsan we need to make sure that the lock is released first, and THEN let "connection" go out of scope.